File: programmers/SummerWinter Coding(~2018)/기지국 설치.js
function solution(n, stations, w) {
  let answer = 0;
  const range = 2 * w + 1;
  const l = stations.length;
  for (let i = 0; i <= l; i++) {
    const s = i == 0 ? 1 : stations[i - 1] + w + 1;
    const e = i == l ? n : stations[i] - w - 1;
    if (s <= e) {
      const length = e - s + 1;
      answer += Math.ceil(length / range);
    }
  }
  return answer;
}

File: programmers/SummerWinter Coding(~2018)/기지국 설치.py
def solution(n, stations, w):
    ans = 0
    l = len(stations)
    idx = 0
    now = 1
    while now <= n:
        if idx < l and now >= stations[idx] - w:
            now = stations[idx] + w + 1
            idx += 1
        else:
            now += 2 * w + 1
            ans += 1
    return ans